File: src/main/java/com/yourwatchrental/watchrental/user/dto/response/UserResponseDTO.java
package com.yourwatchrental.watchrental.user.dto.response;

import com.yourwatchrental.watchrental.user.Role;
import com.yourwatchrental.watchrental.user.User;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

public record UserResponseDTO (
    UUID id,
    String firstName,
    String lastName,
    LocalDate dateOfBirth,
    String email,
    String phoneNumber,
    LocalDateTime createdAt,
    Role role
) { }

File: frontend/src/app/core/models/profile/user-response.dto.ts
import { role } from "./enums/role";
import { userStatus } from "./enums/user-status";

export interface UserResponseDTO{
    id: string;
    firstName: string;
    lastName: string;
    dateOfBirth: string;
    email: string
    phoneNumber: string;
    createdAt: string;
    role: role;
    status: userStatus;
}